capstone-frontend/src/pages/BorderPage.jsx

```diff
@@ -1,101 +1,105 @@
 import Header from "../components/header";
 import { useParams } from "react-router-dom";
-import { useState, useEffect, act } from "react";
+import { useState, useEffect } from "react";
 import "./BorderPage.css";
+import useSWR from "swr";
+const fetcher = (url) => fetch(url).then((res) => res.json());
+import ModalPopulate from "../components/modal";
 
 function BorderPage() {
   const borderObject = useParams();
   const borderIndex = borderObject.borderid;
-  let specificBorderPosts = [];
 
-  const [crossingData, setCrossingData] = useState([]);
   const [postData, setPostData] = useState({
-    userId: "",
-    borderNum: parseInt(borderIndex),
+    userId: "erramoseduardo@gmail.com",
+    borderNum: "0",
     userInput: "",
   });
 
-  useEffect(() => {
-    fetchComments();
-    fetch("http://localhost:5000/borderdata")
-      .then((response) => response.json())
-      .then((data) => {
-        setCrossingData(data);
-      });
-  }, []);
+  const { data, error, isLoading } = useSWR(
+    "http://localhost:5000/borderdata",
```

Comment:
So if I understand correctly, it seems like you're fetching the data for *all* borders here even though you're only displaying the data for *one* border on this page?

That seems like it will potentially slow things down. Perhaps you could fetch all the border data in a component higher up and pass the relevant info into this component? Or only fetch a single border data in this component (although I don't think your backend is set up for that right now)